genmc-sys/src/genmc.rs

```diff
@@ -0,0 +1,39 @@
+use std::sync::OnceLock;
+
+use cxx::UniquePtr;
+
+use super::{GenmcParams, LogLevel, MiriGenmcShim};
+
+static GENMC_LOG_LEVEL: OnceLock<LogLevel> = OnceLock::new();
+
+#[derive(Clone, Copy)]
+/// This struct is used to prevent unsafe access to GenMC.
+/// Any operations that create a GenMC object or call a GenMC function should go through this struct.
+pub struct Genmc {}
+
+impl Genmc {
+    pub fn new(genmc_log_level: LogLevel) -> Self {
```

Comment:
Seems like this function will panic if the 2nd call uses a different log level; please document that.